src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/WPFAddItemCommandHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Input;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    [ExportCommandGroup(CommandGroup.WPF)]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal partial class WPFAddItemCommandHandler : AbstractAddItemCommandHandler
+    {
+        private static readonly Guid s_wpfPackage = new Guid("{b3bae735-386c-4030-8329-ef48eeda4036}");
+
+        protected override Dictionary<long, List<TemplateDetails>> GetTemplateDetails() => new Dictionary<long, List<TemplateDetails>>
+        {
+           // Command Id                                Capability                      ExtraCapability         DirNamePackageGuid          DirNameResourceId                                           TemplateName    TemplateNameResourceId
+           //                                                                                                                                                                   PackageGuid
+            { WPFCommandId.AddWPFWindow,                ProjectCapability.CSharp,       ProjectCapability.WPF,  LegacyCSharpPackageGuid,    LegacyCSharpStringResourceIds.IDS_TEMPLATE_DIRLOCALITEMS,   s_wpfPackage,   WPFTemplateNames.WPFWindow              },
```

Comment:
I don't understand what I'm looking at, where is this ExtraCapability stored?

Reply:
As discussed, its just a convenience of the helper method, to make the table look nicer. The two get combined (with &) in the helper here: https://github.com/dotnet/project-system/pull/4749/files#diff-aa7528047255ba61bc869434cf8fd4a7R18